examples/IMUKittiExampleGPS.cpp

```diff
@@ -0,0 +1,293 @@
+
+    Vector6 BodyP = (Vector(6) << BodyPtx, BodyPty, BodyPtz, BodyPrx, BodyPry, BodyPrz).finished();
+    auto body_T_imu = Pose3::Expmap(BodyP);
+    if (!body_T_imu.equals(Pose3(), 1e-5)) {
+        printf("Currently only support IMUinBody is identity, i.e. IMU and body frame are the same");
+        exit(-1);
+    }
+
+    // Read IMU data
+    // Time dt accelX accelY accelZ omegaX omegaY omegaZ
+    // 46534.47837579 46534.47837579 1.7114864219577 0.1717911743144 9.80533438749 -0.0032006241515747 0.031231284764596 -0.0063569265706488
+    vector<imuMeasurement_t> IMU_measurements;
+    string IMU_data_file = findExampleDataFile("KittiEquivBiasedImu.txt");
+
+    printf("-- Reading IMU measurements from file\n");
+    {
+        ifstream IMU_data(IMU_data_file.c_str());
+        getline(IMU_data, line, '\n'); // ignore the first line
+
+        double time = 0, dt = 0, acc_x = 0, acc_y = 0, acc_z = 0, gyro_x = 0, gyro_y = 0, gyro_z = 0;
+        while (!IMU_data.eof()) {
+            getline(IMU_data, line, '\n');
+            sscanf(line.c_str(), "%lf %lf %lf %lf %lf %lf %lf %lf", &time, &dt, &acc_x, &acc_y, &acc_z, &gyro_x,
+                   &gyro_y, &gyro_z);
+
+            imuMeasurement_t measurement;
+            measurement.Time = time;
+            measurement.dt = dt;
+            measurement.Accelerometer = Vector3(acc_x, acc_y, acc_z);
+            measurement.Gyroscope = Vector3(gyro_x, gyro_y, gyro_z);
+            IMU_measurements.push_back(measurement);
+        }
+    }
+
+    // Read GPS data
+    // Time,X,Y,Z
+    // 46534.478375790000428,-6.8269361350059405424,-11.868164241239471224,0.040306091310000624617
+    vector<gpsMeasurement_t> GPS_measurements;
+    string GPS_data_file = findExampleDataFile("KittiGps_converted.txt");
+
+    printf("-- Reading GPS measurements from file\n");
+    {
+        ifstream GPS_data(GPS_data_file.c_str());
+        getline(GPS_data, line, '\n'); // ignore the first line
+
+        double time = 0, gps_x = 0, gps_y = 0, gps_z = 0;
+        while (!GPS_data.eof()) {
+            getline(GPS_data, line, '\n');
+            sscanf(line.c_str(), "%lf,%lf,%lf,%lf", &time, &gps_x, &gps_y, &gps_z);
+
+            gpsMeasurement_t measurement;
+            measurement.Time = time;
+            measurement.Position = Vector3(gps_x, gps_y, gps_z);
+            GPS_measurements.push_back(measurement);
+        }
+    }
+
+    // Configure different variables
+    double tOffset = GPS_measurements[0].Time;
+    size_t firstGPSPose = 1;
+    size_t GPSskip = 10; // Skip this many GPS measurements each time
+    double g = 9.8;
+    auto w_coriolis = Vector3(); // zero vector
+
+    // Configure noise models
+    noiseModel::Diagonal::shared_ptr noiseModelGPS = noiseModel::Diagonal::Precisions((Vector(6) << Vector3::Constant(0), Vector3::Constant(1.0/0.07)).finished());
```

Comment:
auto

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc

Reply:
Addressed in b48bdfdc696328be3fedfe11db18b83d3341e7bc